modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py

```diff
@@ -67,7 +67,7 @@ def list_of_ips(self):
         result = [None] * len(self.list_of_block_partitions)
         for idx, partition in enumerate(self.list_of_block_partitions):
             partition.drain_call_queue()
-            result[idx] = partition._ip_cache
+            result[idx] = partition.ip(materialize=False)
```

Comment:
What was wrong with getting a bunch of `None`s here? Previously, if the IP cache was not available, this code would do nothing, and now it submits a bunch of kernels to the execution engine, which in theory, can affect performance. How often do we really access the `.ip` field of partitions anyway, do we really need to care about it being up-to-date?

Reply:
`list_of_ips` is called only when the user explicitly requests them via `get_ip` parameter. I think it's strange to return a bunch of `None`s when the user needs it and when it can be calculated (even if it's slower). In general, this functionality (`ip` function) is used quite rarely, it seems only for the `unwrap` function.

https://github.com/modin-project/modin/blob/ce54013f2116fcc57e7a6a09cfdd3532beb24d45/modin/core/dataframe/base/partitioning/axis_partition.py#L195